packages/xrpc-server/src/auth.ts

```diff
@@ -65,6 +72,12 @@ export const verifyJwt = async (
   if (parts.length !== 3) {
     throw new AuthRequiredError('poorly formatted jwt', 'BadJwt')
   }
+
+  const header = parseHeader(parts[0])
+  if (header['typ'] !== 'JWT') {
+    throw new AuthRequiredError('jwt is not of type "JWT"', 'BadJwtType')
+  }
```

Comment:
I don't think this should cause any issues with our implementation since we've set this for some time, but also not sure that it's necessary.  Is there a particular reason it was introduced here?  It's not currently explicitly referenced in the spec: https://atproto.com/specs/xrpc#inter-service-authentication-jwt

Reply:
If it's serving some useful purpose to disambiguate from access/refresh tokens, perhaps a middle-ground is to loosen this a little bit and assume it is a JWT if the claim is missing.

Reply:
This was done with the same philosophy as this:
https://datatracker.ietf.org/doc/html/rfc9449#section-11.5

I guess that now that service auth tokens contain an `lxm`, they are less likely to be confused for something else but I think that when it comes to security, the more constrained the rules, the better.

Reply:
My only concern is that we're making some subtle changes that affect the protocol specs, and we may want to go through a bit more process/discussion before making changes here.  For example, what if we decide we want to use a different value than `JWT`?  If nothing else in this work strictly relies on this change I wonder if we should table it temporarily.

Reply:
Well, a missing `typ` could arguably be interpreted as meaning `"JWT"`:
https://datatracker.ietf.org/doc/html/rfc7519#section-5.1